components/shopify/actions/create-product/create-product.mjs

```diff
@@ -0,0 +1,108 @@
+import shopify from "../../shopify.app.js";
+
+export default {
+  key: "shopify-create-product",
+  name: "Create Product",
+  description: "Create a new product",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    shopify,
+    title: {
+      propDefinition: [
+        shopify,
+        "title",
+      ],
+    },
+    bodyHtml: {
+      type: "string",
+      label: "Body <HTML>",
+      description: `A description of the product
+        Supports HTML formatting
+        Example: \`<strong>Good snowboard!</strong>\``,
+      optional: true,
+    },
+    vendor: {
+      type: "string",
+      label: "Vendor",
+      description: "The name of the product's vendor",
+      optional: true,
+    },
+    productType: {
+      type: "string",
+      label: "Product Type",
+      description: "A categorization for the product used for filtering and searching products",
+      optional: true,
+    },
```

Comment:
Shopify seems to use GraphQL for showing the interactive dropdown menu. It has categories and submenus until a type is selected. Using async options would be different, it would list all "primitive" types and probably be a too big of a list